libraries/shared/watcher/storage_watcher_test.go

```diff
@@ -47,181 +49,341 @@ var _ = Describe("Storage Watcher", func() {
-			go storageWatcher.Execute(diffs, errs, time.Hour)
+			err := storageWatcher.Execute(time.Hour)
 
-			Eventually(func() (string, error) {
-				logContent, err := ioutil.ReadFile(tempFile.Name())
-				return string(logContent), err
-			}).Should(ContainSubstring(fakes.FakeError.Error()))
-			close(done)
+			Expect(err).To(MatchError(fakes.FakeError))
+			logContent, readErr := ioutil.ReadFile(tempFile.Name())
+			Expect(readErr).NotTo(HaveOccurred())
+			Expect(string(logContent)).To(ContainSubstring(fakes.FakeError.Error()))
 		})
 
 		Describe("transforming new storage diffs from csv", func() {
 			BeforeEach(func() {
 				mockFetcher.DiffsToReturn = []utils.StorageDiff{csvDiff}
 				storageWatcher = watcher.NewStorageWatcher(mockFetcher, test_config.NewTestDB(test_config.NewTestNode()))
+				storageWatcher.HeaderRepository = mockHeaderRepository
 				storageWatcher.Queue = mockQueue
 				storageWatcher.AddTransformers([]transformer.StorageTransformerInitializer{mockTransformer.FakeTransformerInitializer})
 			})
 
-			It("executes transformer for recognized storage diff", func(done Done) {
-				go storageWatcher.Execute(diffs, errs, time.Hour)
-
-				Eventually(func() utils.StorageDiff {
-					return mockTransformer.PassedDiff
-				}).Should(Equal(csvDiff))
-				close(done)
-			})
-
-			It("queues diff for later processing if transformer execution fails", func(done Done) {
-				mockTransformer.ExecuteErr = fakes.FakeError
-
-				go storageWatcher.Execute(diffs, errs, time.Hour)
-
-				Expect(<-errs).To(BeNil())
-				Eventually(func() bool {
-					return mockQueue.AddCalled
-				}).Should(BeTrue())
-				Eventually(func() utils.StorageDiff {
-					return mockQueue.AddPassedDiff
-				}).Should(Equal(csvDiff))
-				close(done)
+			Describe("when getting header succeeds", func() {
+				It("executes transformer for recognized storage diff with matching header", func(done Done) {
+					go func() {
+						err := storageWatcher.Execute(time.Hour)
+						Expect(err).NotTo(HaveOccurred())
+					}()
+
+					Eventually(func() utils.StorageDiff {
+						return mockTransformer.PassedDiff
+					}).Should(Equal(csvDiff))
+					close(done)
+				})
+
+				It("recognizes header match even if hash hex missing 0x prefix", func(done Done) {
+					mockHeaderRepository.GetHeaderReturnHash = fakes.FakeHash.Hex()[2:]
+
+					go func() {
+						err := storageWatcher.Execute(time.Hour)
+						Expect(err).NotTo(HaveOccurred())
+					}()
+
+					Eventually(func() utils.StorageDiff {
+						return mockTransformer.PassedDiff
+					}).Should(Equal(csvDiff))
+					close(done)
+				})
+
+				It("queues diff for later processing if transformer execution fails", func(done Done) {
+					mockTransformer.ExecuteErr = fakes.FakeError
+
+					go func() {
+						err := storageWatcher.Execute(time.Hour)
+						Expect(err).NotTo(HaveOccurred())
+					}()
+
+					Eventually(func() utils.StorageDiff {
+						return mockQueue.AddPassedDiff
+					}).Should(Equal(csvDiff))
+					close(done)
```

Comment:
When I looked at this test quickly it took me a minute to understand why that when execution fails `Execute` doesn't isn't expected to return an error. Not sure it's necessary (maybe I should have just looked at the test closer 😜 ) - but do you think it's worth it to test that the transformer execution logged the error?

Reply:
👍 e5973ea

Not returning an error via `Execute` is a bit confusing, but I think it makes sense to continue in most failure scenarios. Definitely up for returning errors if we consider anything other than a fetcher error a showstopper